### PR TITLE
Add attr for recursively setting up git submodules

### PR DESCRIPTION
**Pull request by keith**
If a repo has multiple levels of submodules this hydrates those.

### CODE REVIEW

**Comment by keith** [comment]
If we're not ok with this being a breaking change maybe we can provide another attribute like `recursively_init_submodules` for this?

**Comment by jin** [comment]
> If we're not ok with this being a breaking change maybe we can provide another attribute like recursively_init_submodules for this?

Yes, I'd prefer that this is an attribute on `git_repository`.

**Comment by keith** [comment]
Updated with that approach!

**Comment by jin** [comment]
Could you add a test for this in src/test/shell/bazel/skylark_git_repository_test.sh?

**Comment by keith** [comment]
I added a test that uses the existing fixtures, so it doesn't actually verify the recursion, just that any are setup, I'm happy to add a new fixture for that if I should?

**Comment by jin** [comment]
Sorry for the delay, yes, we should have a fixture that sets up and verifies the `--recursive` flag.

**Comment by jin** [comment]
Please add a test with a recursive fixture, and I'll import this. Thanks!

**Comment by keith** [comment]
I spent a while looking into this and it's pretty unclear to me how these fixtures are supposed to work. I'll try to look into it again at some point.

**Comment by keith** [comment]
The test repo stuff here is a little too hard to follow, I can't figure out how it's supposed to work. Can we merge this knowing it doesn't break any existing behavior?

**Comment by jin** [comment]
Hmm, could we at least test that the attribute triggers the `--recursive` argument to show up in the repository rule executed command?

**Comment by keith** [comment]
Is there a starlark test somewhere that has access to that info? All I see is the shell tests where it doesn't seem like I could access the command

**Comment by keith** [comment]
Thanks!
